modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/persistence/PersistentPageMemoryMetricSource.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.pagememory.persistence;
+
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.ignite.internal.metrics.Metric;
+import org.apache.ignite.internal.metrics.MetricSet;
+import org.apache.ignite.internal.metrics.MetricSource;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Metric source for persistent page memory.
+ */
+public class PersistentPageMemoryMetricSource implements MetricSource {
```

Comment:
I would ask you to indicate in the documentation for the fields that they are guarded by the monitor, otherwise I immediately notice bugs.

Reply:
> otherwise I immediately notice bugs.

What do you mean? What bugs?

Reply:
As soon as I saw these fields, I thought there would be data races, until I looked at each method.
Do you think it might be worth using method synchronization instead of synchronization by internal mutex?

Reply:
Honestly, I just did what all other implementations did :)
But I don't mind synchronizing on `this` here, but if someone would insist, I'd fix it. I know that it is not a recommended practice